### PR TITLE
document the jj equivalent of `git reset --soft`

### PR DESCRIPTION
**Pull request by jyn514**
# Checklist

If applicable:
- [x] I have updated `CHANGELOG.md`
- [x] I have updated the documentation (README.md, docs/, demos/)
- [x] I have updated the config schema (cli/src/config-schema.json)
- [x] I have added tests to cover my changes

### CODE REVIEW

File: docs/git-comparison.md [hunk 1]
```diff
@@ -211,6 +211,11 @@ parent.
       <td><code>git reset --hard</code> (same as abandoning a change since Git
           has no concept of a "change")</td>
     </tr>
+    <tr>
+      <td>Abandon the current change, but keep its diff in the working copy</td>
+      <td><code>jj move --from @- --to @</code></td>
```
Comment (martinvonz) [inline docs/git-comparison.md:216]: nit: `--to @` is the default when `--from` is used, so I would leave that out

File: docs/git-comparison.md [hunk 2]
```diff
@@ -211,6 +211,11 @@ parent.
       <td><code>git reset --hard</code> (same as abandoning a change since Git
           has no concept of a "change")</td>
     </tr>
+    <tr>
+      <td>Abandon the current change, but keep its diff in the working copy</td>
```
Comment (martinvonz) [inline docs/git-comparison.md:215]: It's the *parent* change we abandon here, so I would say "Abandon the parent change" here. I hope that's not too confusing.

Btw, Mercurial, which has no automatic snapshotting of the working copy as a commit, also calls it "the parent of the working copy" :) I remember thinking that term was confusing, but it now makes a lot of sense to me.

File: docs/git-comparison.md
Comment (martinvonz) [inline docs/git-comparison.md:1]: > i want to undo the most recent commit

A bit off topic, but it sounds like you could have used `jj undo` in your case?

File: docs/git-comparison.md (hunk 2 above)
Comment (jyn514) [reply docs/git-comparison.md:215]: i think "parent of the working copy" is great phrasing, since we want this to also apply for git :) i'll use that.

**Review by martinvonz — APPROVED** [review]

**Pull request merged**